### ISSUE TITLE
Advertise ZFSBootMenu as mode of choice for booting from ZFS

### ISSUE THREAD

**Issue by GregorKopka** [openzfs/zfs#14485]
### Describe the feature would like to see added to OpenZFS
Documentation advertising https://ZFSBootMenu.org as *the* method for booting from ZFS on x86, as a start.

### How will this feature improve OpenZFS?
What problem does this feature solve?
GRUB.

- Removes all GRUB-induced limits on pool layout for root pools
- Easy management of boot environments 
- Simple and clean setup, upgrades and disaster recovery

### Additional context
Given the amount of time being spent on assisting with problems around booting, the mess created by some distributions to create their own flavor of dealing with these problems (like not being easily human readable, see the approach of ubuntu)... that is lifetime that could be rescued to be spent on better things.

Auto building a drop-in version of zfsbootmenu.EFI (with current kernel, so new hardware would eg. have working keyboards - which for my new notebook some current releases did not have) for new zfs releases would also remove any and all future problems related to on-disk format changes that ave repeatedly stopped other boot methods (\*cough\* GRUB \*cough\*) dead in their tracks.

**Comment by Majiir** [comment]
If there are boot loaders or distros that are challenging to use with ZFS, then we can document those specific cases and common pitfalls. Documentation should help users use ZFS with their chosen distro or boot loader as best possible. I don't think we should proclaim one boot loader as "_the_ method" just because GRUB+Ubuntu is frustrating.

Pragmatically speaking, declaring ZFSBootMenu "_the_ method" risks confusing users into thinking ZFS may _only_ be used on root with ZFSBootMenu.

I run root on ZFS on three devices with three different boot loaders, none of which is ZFSBootMenu. Other boot loaders are perfectly capable.

**Comment by GregorKopka** [comment]
It's not Ubuntu+x - but **+GRUB**, which is annoying.

**Comment by eirnym** [comment]
@Majiir I agree with you that this is not _"the only"_ method to boot ZFS root, and yes, all operations presented in this were known before, especially in FreeBSD and Solaris/Illumos. 

What ZFSBootMenu brings is a boot menu which works out of the box and easy to use, which is a quite impressive, it's safe when you accidentially update a kernel in your main OS thanks to to ZFS Boot Environments technique, and works with native zfs encryption. 

Why does it works well? it's a stable linux kernel + minimal set of tools to support this functionality.

As far as I know, one of the main goals of GRUB is be as system-agnostic as possible, as minimalistic as possible and to be able to run on the slowest and limited hardware as possible. Thus, it contains a its own FS parsers, from which it has all limitations it has.

FreeBSD boot loader (stage 2) has been very small as well before ZFS came into FreeBSD, later it has been extended to support ZFS and rewritten to be nice and easy to use. 

On my personal opinion, ZFSBootMenu endorsement from OpenZFS project will be a good idea to popularize filesystem across Linux users even more.

**Comment by Majiir** [comment]
> What bootloaders do you use and what is their feature set in comparison?

My current preference is systemd-boot. I also use U-Boot for an ARM box, and I have one machine still running GRUB. I run NixOS, which obviates most of the benefits of ZFSBootMenu, since NixOS has its own approach to system rollbacks. To the example from above, if I install a bad kernel or break my initrd, I can boot the previous config without a ZFS rollback.

I don't object to endorsing what looks like a fine bootloader option. I do object to the framing that it's the only choice worth endorsing or supporting, or even that it should be promoted as the number-one choice. ZFSBootMenu does not support ARM, it may not be as easy to install or configure on a user's distro as other bootloaders, it may pose challenges for secure boot setups, etc.

IMHO, it will help the largest number of users if the ZFS documentation warns users about footguns with GRUB, recommends multiple bootloaders that work well, and helps users get ZFS running with their bootloader of choice.
